### DISCUSSION TITLE
Parameters num_samples, top_k, top_p, temperature affect the fine tuning process?

### DISCUSSION THREAD

**Discussion by carlos-ariza3** [amazon-science/chronos-forecasting#176] [Q&A] [answered]
Hi, I want to know if the modification of the parameters **num_samples**, **top_k**, **top_p** and **temperature** affect the fine tuning process. All those parameters can be modified in the pipeline.predict() function, so, I'm not sure if there is any difference by modifying them in the chronos_config yaml file (chronos-t5-small.yaml). 
Also, I'm not sure if those parameters affect the fine tuning because **num_samples** appears in the chronos-t5-small.yaml but **top_k**, **top_p** and **temperature** don't appear in this file, but, all those are parameters in the main() function of [train.py](https://github.com/amazon-science/chronos-forecasting/blob/main/scripts/training/train.py#L668) (lines 536-539).
Thanks.

**Comment by lostella** [comment, marked as answer]
Hi @carlos-ariza3, the `num_samples`, `top_k`, `top_p`, and `temperature` parameters only affect the model's predictions, and **have no impact on training or fine-tuning**.

The reason they appear in the training script is that they are needed to instantiate a `ChronosConfig` object, which we serialize together with the model. At prediction time, the values of these parameters will be the "defaults", which can be overridden by passing new values to `pipeline.predict(...)`.

**Reply by carlos-ariza3** [reply]
Thank you very much!

**Comment by lostella** [comment]
Hi, the docstring of the predict method explains how to use them, see https://github.com/amazon-science/chronos-forecasting/blob/ac6ee36acee1e47446cd66f72f540c87f1f1fbe4/src/chronos/chronos.py#L467 You can pass them as keyword arguments.Am 28.10.2024 um 11:13 schrieb Jadin C Jackson, PhD ***@***.***>:﻿
Hi Lorenzo,
Would you be able to either provide a link to documentation on how to pass in those values as parameters to the pipeline.predict() method? If no documentation exists, could to provide some examples here?  e.g. are they simply passed in as keyword arguments?
pipeline.predict(num_samples = int, top_k = int, top_p = int and temperature = float, ...)
